irohad/ordering/impl/kick_out_proposal_creation_strategy.cpp

```diff
@@ -0,0 +1,59 @@
+#include <utility>
+
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ordering/impl/kick_out_proposal_creation_strategy.hpp"
+
+#include <algorithm>
+#include "interfaces/common_objects/peer.hpp"
+
+using namespace iroha::ordering;
+
+KickOutProposalCreationStrategy::KickOutProposalCreationStrategy(
+    std::shared_ptr<SupermajorityCheckerType> majority_checker)
+    : majority_checker_(std::move(majority_checker)) {}
+
+void KickOutProposalCreationStrategy::onCollaborationOutcome(
+    const PeerList &peers) {
+  std::lock_guard<std::mutex> guard(mutex_);
+  RoundCollectionType last_requested;
+  for (const auto &peer : peers) {
+    auto iter = last_requested_.find(peer);
+    if (iter != last_requested_.end()) {
+      last_requested.insert(*iter);
+    } else {
+      last_requested.insert({peer, RoundType{0, 0}});
+    }
+  }
+  last_requested_ = last_requested;
+}
+
+void KickOutProposalCreationStrategy::shouldCreateRound(
+    RoundType round, const std::function<void()> &on_create) {
```

Comment:
It is preferable not to use std::function in non-tests code due to its performance drawbacks.

Reply:
What is your suggestion to use instead?

Reply:
We agreed to leave it as is during an offline discussion.